update-pot

```diff
@@ -25,24 +25,24 @@ check_canaries() {
 HERE="$(readlink -f "$(dirname "$0")")"
 
 OUTPUT="$HERE/po/snappy.pot"
-if [ -n "$1" ]; then
+if [ -n "${1:-}" ]; then
 	OUTPUT="$1"
 fi
 
 # ensure we have our xgettext-go
 go install github.com/snapcore/snapd/i18n/xgettext-go
 
+tmpdir="$(mktemp -d)"
+trap 'rm -rf "$tmpdir"' EXIT
+
 # exclude vendor and _build subdir
-I18N_FILES="$(mktemp -d)/i18n.files"
-find "$HERE" -type d \( -name "vendor" -o -name "_build" -o -name ".git" \) -prune -o -name "*.go" -type f -print > "$I18N_FILES"
-# shellcheck disable=SC2064
-trap "rm -rf $(dirname "$I18N_FILES")" EXIT
+find "$HERE" -type d \( -name "vendor" -o -name "_build" -o -name ".git" \) -prune -o -name "*.go" -type f -printf "%P\n" > "$tmpdir/go.files"
```

Comment:
Nice use of `-prune`

Reply:
I can't take credit for it: that was introduced by @mvo in 117734ca395.